fog/distribution/src/main.rs

```diff
@@ -318,14 +281,57 @@ fn main() {
     thread::sleep(Duration::from_secs(1));
 }
 
+fn build_fog_resolver(
+    fog_uri: &FogUri,
+    env: &Arc<grpcio::Environment>,
+    logger: &Logger,
+) -> FogResolver {
+    // Ensure there are fog reports available
+    // XXX: This retry should possibly be in the GrpcFogPubkeyResolver object itself
+    // instead 15'th fibonacci is 987, so the last delay should be ~100
+    // seconds
+    let conn = GrpcFogReportConnection::new(env.clone(), logger.clone());
+    let responses = retry(
+        delay::Fibonacci::from_millis(100)
+            .map(delay::jitter)
+            .take(15),
+        || match conn.fetch_fog_reports(core::slice::from_ref(fog_uri).iter().cloned()) {
+            Ok(responses) => OperationResult::Ok(responses),
+            Err(ReportConnError::Rpc(err)) => {
+                log::error!(
+                    logger,
+                    "grpc error reaching fog report server, retr700/751:ying: {}",
+                    err
+                );
+                OperationResult::Retry(ReportConnError::Rpc(err))
+            }
+            Err(ReportConnError::NoReports(_)) => {
+                log::error!(logger, "no fog reports available, retrying");
+                OperationResult::Retry(ReportConnError::NoReports(fog_uri.clone()))
+            }
+        },
+    )
+    .expect("Could not contact fog report server");
+
+    let report_verifier = {
+        let mr_signer_verifier = fog_ingest_enclave_measurement::get_mr_signer_verifier(None);
+        let mut verifier = Verifier::default();
+        verifier.debug(DEBUG_ENCLAVE).mr_signer(mr_signer_verifier);
+        verifier
+    };
+
+    FogResolver::new(responses, &report_verifier).expect("Could not get FogResolver")
+}
+
 fn worker_thread_entry(
     spendable_txouts_receiver: crossbeam_channel::Receiver<SpendableTxOut>,
     running_threads_sender: crossbeam_channel::Sender<usize>,
     config: Config,
     ledger_db: LedgerDB,
     fog_accounts: Vec<AccountKey>,
-    fog_resolver: FogResolver,
+    mut fog_resolver: FogResolver,
```

Comment:
Why does this needs to be passed as a parameter if the method can create it on its own? Any reason not to have this function create it before the loop?

Reply:
i advocated for this to be the current way, the reasoning is, if it happens before `worker_thread_entry`, probably it happens once and all the threads can use it. if it happens always at the start of `worker_thread_entry` then all the threads have to hit the report server at the beginning

it probably doesn't make too much difrerence

Reply:
More load on the report server is not a bad thing to test for :)

Reply:
yeah that's fair